src/backend/distributed/cdc/Makefile

```diff
@@ -1,15 +1,38 @@
-MODULE_big = citus_pgoutput
-EXTENSION = citus_pgoutput
+ifneq ($(DECODER),)
+    DECODER_NAME=$(DECODER)
+else
+	DECODER_NAME = pgoutput
+	DECODER = pgoutput
+endif
 
+citus_subdir = src/backend/distributed/cdc
+citus_top_builddir = ../../../..
 
-OBJS = cdc_decoder.o cdc_decoder_utils.o
+MODULE_big = citus_${DECODER_NAME}
+EXTENSION = citus_${DECODER_NAME}
 
+.PHONY: clean-full install install-all
 
+OBJS = cdc_decoder.o cdc_decoder_utils.o
 SHLIB_LINK = $(libpq)
 
 PG_CONFIG = pg_config
 PGXS := $(shell $(PG_CONFIG) --pgxs)
 INCDIR := $(shell $(PG_CONFIG) --includedir)
 include $(PGXS)
 
-override CFLAGS += -std=gnu99 -Wall -Wextra -Wno-unused-parameter -Wno-sign-compare -Wno-missing-field-initializers -Wno-clobbered -Wno-declaration-after-statement -Wendif-labels -Wmissing-format-attribute -Wmissing-declarations -Wmissing-prototypes -Wshadow -Werror=vla -Werror=implicit-int -Werror=implicit-function-declaration -Werror=return-type -fstack-clash-protection
\ No newline at end of file
+override CFLAGS += -std=gnu99 -Wno-declaration-after-statement
+
+clean:
+	rm -f *.o *.so
+
+cleanup-before-install:
+	rm -f $(DESTDIR)$(datadir)/$(datamoduledir)/citus_$(DECODER).control
+	rm -f $(DESTDIR)$(datadir)/$(datamoduledir)/citus-$(DECODER).so
+	rm -f $(DESTDIR)$(datadir)/$(datamoduledir)/citus_decoders/$(DECODER).so
+
+install: cleanup-before-install
+	mkdir -p '$(DESTDIR)$(datadir)/$(datamoduledir)/citus_decoders'
+	$(INSTALL_SHLIB) citus_$(DECODER).so '$(DESTDIR)$(datadir)/$(datamoduledir)/citus_decoders/$(DECODER).so'
+	$(INSTALL_DATA) citus_$(DECODER).control '$(DESTDIR)$(datadir)/$(datamoduledir)/citus_decoders/$(DECODER).control'
```

Comment:
do we actually need a .control file? I think this is mainly needed when there are SQL functions / tables /etc and the user needs to do a CREATE EXTENSION.

Reply:
I got an error while installing(make install) without .control file, but will try again

Reply:
OK after removing the EXTENSION from Makfile, the .control files are no longer needed.
I have removed them now..